### PR TITLE
face mask detector v1

### PR DESCRIPTION
**Pull request by IvanKurnosov**
Notable changes:

1. Change numpy version in requirements in oder to compatibility with tensorflow 2.5.0
2. Add mtcnn to requirements. When I specify this requirement only in facenet.__init__.requirements together with tensorflow, i have an keras attribute error when i build docker. The solution is to install mtcnn before tensorflow and i didn't found better way than specify it in main requirements. It is lightweight, but may be there is better solution
3. In managers.PluginManager.__init__ I change the way of plugin path calculating. Before it was cant't process plugins like facenet.facemask.MaskDetector becouse it consider that facenet was the last packege.
4. managers.MLModel._extract - Before all files were extracted to the root. But wen you loading tf.keras model you need to extract zip how it is.

### CODE REVIEW

File: embedding-calculator/README.md [hunk 1]
```diff
@@ -35,9 +35,9 @@ $ docker run -p 3000:3000 exadel/compreface-core:latest
 
 | Tag                    | Scanner     | Build arguments                                                                                          | Comment                            |
 |------------------------|-------------|----------------------------------------------------------------------------------------------------------|------------------------------------|
-| :0.5.1 :latest         | Facenet2018 |                                                                                                          |                                    |
-| :0.5.1-insightface     | InsightFace | FACE_DETECTION_PLUGIN=insightface.FaceDetector<br>CALCULATION_PLUGIN=insightface.Calculator              |                                    |
-| :0.5.1-insightface-gpu | InsightFace | FACE_DETECTION_PLUGIN=insightface.FaceDetector<br>CALCULATION_PLUGIN=insightface.Calculator<br>GPU_IDX=0 | CORE_GPU_IDX - index of GPU-device |
+| :0.5.0 :latest         | Facenet2018 |                                                                                                          |                                    |
```
Comment (pospielov) [inline embedding-calculator/README.md:38]: Why did you change the version?

File: embedding-calculator/requirements.txt [hunk 2]
```diff
@@ -16,12 +16,16 @@ pylama~=7.7.1
 # dependencies for both scanner backends
 Pillow~=8.0.1
 imagecodecs~=2020.5.30
-numpy~=1.18.0
+#numpy~=1.18.0
```
Comment (pospielov) [inline embedding-calculator/requirements.txt:19]: remove comment

File: embedding-calculator/src/services/facescan/plugins/base.py [hunk 3]
```diff
@@ -71,11 +71,19 @@ def _download(cls, url: str, output):
     def _extract(self, filename: str):
         os.makedirs(self.path, exist_ok=True)
         with ZipFile(filename, 'r') as zf:
-            for info in zf.infolist():
-                if info.is_dir():
-                    continue
-                file_path = Path(self.path) / Path(info.filename).name
-                file_path.write_bytes(zf.read(info))
+            if str(self.path)[-18:] == 'face_mask_detector':
```
Comment (pospielov) [inline embedding-calculator/src/services/facescan/plugins/base.py:74]: Add a parameter "retain_folder_structure" to BasePlugin, and add it to every model initialization instead of this.
And change the behavior using this variable

File: embedding-calculator/src/services/facescan/plugins/facenet/facemask/facemask.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+from src.services.dto import plugin_result
+
+import cv2
+
+
+class MaskDetector(base.BasePlugin):
+    slug = 'mask'
+    LABELS = ('without_mask', 'with_mask', 'mask_weared_incorrect')
+    ml_models = (
+        ('face_mask_detector', '1AeSYb_E_3cqZM67qXnJ__wJDgw6yqTDV'),
```
Comment (pospielov) [inline embedding-calculator/src/services/facescan/plugins/facenet/facemask/facemask.py:33]: Name of the model architecture and probably dataset, instead of 'face_mask_detector'

File: embedding-calculator/src/services/facescan/plugins/facenet/facemask/facemask.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        )
+
+        def get_value(img: Array3D) -> Tuple[Union[str, Tuple], float]:
+            img = cv2.resize(img, dsize=(self.INPUT_IMAGE_SIZE, self.INPUT_IMAGE_SIZE),
+                             interpolation=cv2.INTER_CUBIC)
+            img = img[:, :, [2, 1, 0]]
+            img = np.expand_dims(img, 0)
+
+            scores = model.predict(img)
+            print('Predictions: ' + str(scores))
```
Comment (pospielov) [inline embedding-calculator/src/services/facescan/plugins/facenet/facemask/facemask.py:54]: Remove

File: embedding-calculator/src/services/facescan/plugins/facenet/facenet.py [hunk 6]
```diff
@@ -54,11 +62,19 @@ class FaceDetector(mixins.FaceDetectorMixin, base.BasePlugin):
     det_threshold_b = 0.7059968943
     det_threshold_c = 0.5506904359
 
-    @cached_property
+    '''    @cached_property
```
Comment (pospielov) [inline embedding-calculator/src/services/facescan/plugins/facenet/facenet.py:65]: Delete it and all other comments

File: embedding-calculator/src/services/facescan/plugins/insightface/insightface.py [hunk 7]
```diff
@@ -61,7 +61,7 @@ class FaceDetector(InsightFaceMixin, mixins.FaceDetectorMixin, base.BasePlugin):
     ml_models = (
         ('retinaface_mnet025_v1', '1ggNFFqpe0abWz6V1A82rnxD6fyxB8W2c'),
         ('retinaface_mnet025_v2', '1EYTMxgcNdlvoL1fSC8N1zkaWrX75ZoNL'),
-        ('retinaface_r50_v1', '1LZ5h9f_YC5EdbIZAqVba9TKHipi90JBj'),
+        ('retinaface_r50_v1', '1hvEv4xZP-_50cO7IYkH6sDUb_SC92wut'),
```
Comment (pospielov) [inline embedding-calculator/src/services/facescan/plugins/insightface/insightface.py:64]: What is changed?

File: embedding-calculator/src/services/facescan/plugins/managers.py [hunk 8]
```diff
@@ -36,7 +36,11 @@ class PluginManager:
     def __init__(self):
         self.plugins_modules = defaultdict(list)
         for plugin_name in self.get_plugins_names():
-            module = import_module(f'{__package__}.{plugin_name.split(".")[0]}')
+            # this don't work's
```
Comment (pospielov) [inline embedding-calculator/src/services/facescan/plugins/managers.py:39]: Delete comment

**Review by pospielov — APPROVED** [review]

**Comment by IvanKurnosov** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by isobolev — APPROVED** [review]
